codegen/config/binder.go

```diff
@@ -476,7 +476,18 @@ func (b *Binder) TypeReference(schemaType *ast.Type, bindTarget types.Type) (ret
 		ref.GO = b.CopyModifiersFromAst(schemaType, ref.GO)
 
 		if bindTarget != nil {
-			if err = code.CompatibleTypes(ref.GO, bindTarget); err != nil {
+			// if the bind type implements the graphql.ContextMarshaler/graphql.ContextUnmarshaler/graphql.Marshaler/graphql.Unmarshaler interface, we can use it
+			if hasMethod(bindTarget, "MarshalGQLContext") && hasMethod(bindTarget, "UnmarshalGQLContext") {
+				ref.IsContext = true
+				ref.IsMarshaler = true
+				ref.Marshaler = nil
+				ref.Unmarshaler = nil
```

Comment:
I'm not sure about these two lines.

Reply:
It's because we may set them on https://github.com/99designs/gqlgen/pull/3632/files#diff-ab508c289540d69c88d9f1146004bbb03b2fbefa9b9ce932e044927fea85aeccL449-L450.

For the `bindTarget` implementing these interface, it's Marshaler and Unmarshaler should be nil.